services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,310 @@
+	}
+	_, err := ctrl.CreateOrUpdate(r.Context(), client, storageClass, mutateFunc)
+	if err != nil {
+		klog.Errorf("failed to create storageClass: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func createCephFilesystemDataPool(w http.ResponseWriter, r *http.Request, client client.Client, poolName, lsoStorageClassName string, dataProtectionPolicy int, enableCompression bool, failureDomain string, storageCluster *ocsv1.StorageCluster) {
+	compression := "none"
+	if enableCompression {
+		compression = "aggressive"
+	}
+
+	datapool := cephv1.NamedPoolSpec{
+		Name: poolName,
+		PoolSpec: cephv1.PoolSpec{
+			FailureDomain: failureDomain,
+			DeviceClass:   lsoStorageClassName,
+			Replicated: cephv1.ReplicatedSpec{
+				Size:                   uint(dataProtectionPolicy),
+				RequireSafeReplicaSize: true,
+			},
+			Parameters: map[string]string{
+				"compression_mode": compression,
+			},
+			EnableCrushUpdates: true,
+		},
+	}
+
+	mutateFunc := func() error {
+		storageCluster.Spec.ManagedResources.CephFilesystems.AdditionalDataPools = append(storageCluster.Spec.ManagedResources.CephFilesystems.AdditionalDataPools, datapool)
+		return nil
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.Context(), client, storageCluster, mutateFunc)
+	if err != nil {
+		klog.Errorf("failed to update storageCluster: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func createCephFilesystemStorageClass(w http.ResponseWriter, r *http.Request, client client.Client, storageClassName, poolName, reclaimPolicy, volumeBindingMode string, filesystemName, namespace string) {
+	allowVolumeExpansion := true
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: storageClassName,
+			Annotations: map[string]string{
+				"description": "Provides RWO and RWX Filesystem volumes",
+			},
+		},
+		Provisioner:       storagecluster.CephFSDriverName,
+		ReclaimPolicy:     (*corev1.PersistentVolumeReclaimPolicy)(&reclaimPolicy),
+		VolumeBindingMode: (*storagev1.VolumeBindingMode)(&volumeBindingMode),
+		// AllowVolumeExpansion is set to true to enable expansion of OCS backed Volumes
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Parameters: map[string]string{
+			"clusterID": namespace,
+			"fsName":    filesystemName,
+			"pool":      poolName,
+			"csi.storage.k8s.io/provisioner-secret-name":            "rook-csi-cephfs-provisioner",
+			"csi.storage.k8s.io/provisioner-secret-namespace":       namespace,
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-cephfs-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-cephfs-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": namespace,
+		},
+	}
+
+	mutateFunc := func() error {
+		return nil
+	}
+	_, err := ctrl.CreateOrUpdate(r.Context(), client, storageClass, mutateFunc)
```

Comment:
Since storageClass params cannot be updated after it is created once, would it make sense to use create here? and if the params differ then delete and create it again?